modules/thermal_hydraulics/include/components/Component.h

```diff
@@ -275,6 +280,18 @@ class Component : public THMObject, public LoggingInterface, public NamingInterf
    */
   virtual void setupMesh() {}
 
+  /**
+   * Method to add a relationship manager for the objects being added to the system. Relationship
+   * managers have to be added relatively early. In many cases before the Action::act() method
+   * is called.
+   *
+   * This method was copied from Action.
```

Comment:
Yeah, this made me uncomfortable - it was just because of a few necessary member data/methods. It should be possible though. The utility function for
```
bool
Action::addRelationshipManager(
    Moose::RelationshipManagerType /*input_rm_type*/,
    const InputParameters & moose_object_pars,
    std::string rm_name,
    Moose::RelationshipManagerType rm_type,
    Moose::RelationshipManagerInputParameterCallback rm_input_parameter_func,
    Moose::RMSystemType)
```
would need to look like
```
bool
addRelationshipManager(
    name,
    factory,
    mesh,
    app,
    Moose::RelationshipManagerType /*input_rm_type*/,
    const InputParameters & moose_object_pars,
    std::string rm_name,
    Moose::RelationshipManagerType rm_type,
    Moose::RelationshipManagerInputParameterCallback rm_input_parameter_func,
    Moose::RMSystemType)
```
Is this preferred?

Reply:
@lindsayad thoughts?

Reply:
Because I'm changing the method to remove the `added` logic, I'll skip the utils functions.

Reply:
yea since they ended up being different, I think not having it is ok. If we see more duplication in the future I think it's a good idea